seer_pas_sdk/core/sdk.py

```diff
@@ -2521,29 +2592,51 @@ def get_volcano_plot_data(
             else:
                 return obj.volcano_plot
 
-    def get_analysis_samples(self, analysis_id: str):
+    def _get_analysis_samples(
+        self, analysis: str, is_name: bool = False, as_df=False
+    ):
         """
-        Get the samples associated with a given analysis ID.
+        Get the samples associated with a given analysis.
 
         Args:
-            analysis_id (str): The analysis ID.
+            analysis (str): The analysis identifier. Accepts an analysis ID or an analysis name.
+            is_name (bool) : Mark true if analysis is an analysis name
```

Comment:
To match the other functions:
```suggestion
            analysis_id (str): Optional analysis identifier.
            analysis_name (str) : Optional analysis name
```
If both are unspecified (null), raise `ValueError`.

Reply:
This is intended to be a helper function, per the underscore prefix. The end user should not be encouraged to use this functionality as we aligned that `get_samples` should be the universal function for retrieving sample data. I chose this interface to support the pattern that I employ for the calling functions.

Reply:
I think the recommended coding pattern is to avoid changing the semantics of one function argument based on the value of another one.